nucypher/utilities/latency.py

```diff
@@ -0,0 +1,78 @@
+import collections
+import time
+from collections import defaultdict
+from threading import Lock
+from typing import List
+
+from eth_typing import ChecksumAddress
+
+
+class NodeLatencyStatsCollector:
+    """
+    Thread-safe utility that tracks latency statistics related to P2P connections with other nodes.
+    """
+
+    MAX_MOVING_AVERAGE_WINDOW = 5
```

Comment:
This is 5 deque values?

Reply:
Yep, only the last 5 values will be used for judging latency. At the moment we calculate average - but that calc can change if we find it not sufficiently representative of actual latency. @cygnusv 's comment here - https://github.com/nucypher/nucypher/pull/3562#discussion_r1830745901

Reply:
My intuition is that this is a pretty small window, but since we're mainly using this calculation for instant consumption, it's a reasonable size. 👍🏻